src/main/java/org/cryptomator/frontend/fuse/ReadOnlyAdapter.java

```diff
@@ -36,13 +37,16 @@
 import java.nio.file.attribute.BasicFileAttributes;
 import java.nio.file.attribute.PosixFileAttributeView;
 import java.nio.file.attribute.PosixFileAttributes;
+import java.text.Normalizer;
 import java.util.Collections;
 import java.util.EnumSet;
 import java.util.Set;
 import java.util.function.BooleanSupplier;
 
 /**
  * Read-Only FUSE-NIO-Adapter based on Sergey Tselovalnikov's <a href="https://github.com/SerCeMan/jnr-fuse/blob/0.5.1/src/main/java/ru/serce/jnrfuse/examples/HelloFuse.java">HelloFuse</a>
+ * <p>
+ * Any (string) path from a fuse callback is normalized into its NFC representation before it is transformed into a {@link java.nio.file.Path}.
```

Comment:
Actually in order to replace iconv, we don't want hard-coded NFC. Both sides need to be configurable.